docs/src/main/asciidoc/_templates/template-concept.adoc

```diff
@@ -32,6 +32,50 @@ include::{includes}/extension-status.adoc[]
 - xref:{doc-guides}/doc-concept.adoc#concept[Quarkus documentation content types: Concept guides]
 - xref:{doc-guides}/doc-reference.adoc[Quarkus style and content guidelines]
 
+== Anchors for in-file and cross-file navigation
+
+* To create an anchored ID, use lower-case characters, Separate each word with `,` and enclose the ID in `[[]]` as in the example below.
++
+[source,asciidoc]
+----
+[[title-heading]]
```

Comment:
To prevent confusion, should we distinguish between the 2 title heading items by adding id to the anchor ID and use that reference elsewhere:

```suggestion
[[title-heading-id]]
```

Reply:
I think it's going to be cumbersome and also a bit counterproductive as if you want consistency, you will have to fix all the existing ones.

Reply:
OK. See your points.

Above implies that the [[bit in here]] matches the title heading. They can be different values. The beauty of using IDs is that the ID can differ and diverge from the original section/topic title without breaking xrefs.  Not a hill to die on though. Thanks.

Reply:
Yes, I agree, but I didn't say that a contributor need to create an ID anchor by verbatim rewrote of a dedicated heading and putting it in the kebab-case. The main point here is that contributors will put it down to **[[]]**, and IF they consider using the heading "shape" as an anchor, I am OK with that since this is so much better than, for example, "section-two-heading-anchor-id", since this doesn't provide the reviewer with any clue to which section contributor was referring to, and is also better for any possible future tweaking/updates if the anchors are made in a very similar way as the text objects they are derived from. Also, how a contributor names its xref link using the **[]-part** is completely up to them, but the ID that works in xref's syntax should be a derivation of the original object.
I spent some time with upstream IDs and their calls lately :), and I can say that I have seen them all, so the bid about their variation and how they work is clear to me :)  haha. 
I can't tell anybody how to contribute, but if there are some barriers in which a contributor can contribute and respect these just a little, our job is done.